CONTRIBUTING.md

```diff
@@ -0,0 +1,368 @@
+
+Build time:   2017-12-06 09:05:06 UTC
+Revision:     cf7821a6f79f8e2a598df21780e3ff7ce8db2b82
+
+Groovy:       2.4.12
+Ant:          Apache Ant(TM) version 1.9.9 compiled on February 2 2017
+JVM:          1.8.0_212-release (JetBrains s.r.o 25.212-b4-5784211)
+OS:           Mac OS X 10.14.6 x86_64
+```
+
+Do you have the Android SDK setup, and do you have a pointer to the Java environment?
+
+```
+echo -e $ANDROID_HOME\\n$JAVA_HOME 
+/Users/jhwill/Library/Android/sdk
+/Applications/Android Studio.app/Contents/jre/jdk/Contents/Home
+```
+
+### Problems with the Build
+
+Take a look at the build instructions in the project's [README.md](https://github.com/aws-amplify/aws-sdk-android#building-the-sdk).
+
+If the build fails, and you can't figure out why from a Google search /
+StackOverflow session, try passing options to Gradle:
+
+```
+./gradlew --stacktrace
+```
+
+The next flag will spit out lots of info. It's only useful if you pipe the
+output to a file, and grep through it.
+
+```
+./gradlew --debug 2>&1 > debugging-the-build.log
+```
+
+### Failing Instrumentation Tests
+
+If a single test is failing, run only that test, to isolate it. You may also
+want to see the output on the device that occurs, while the tests are
+executing.
+
+```bash
+# If you run this same code in a script/block, this will clear the log
+# file each time.
+rm -f device-logs-during-test.log
+
+# Clear the device's, so you only get recent stuff.
+adb logcat -c
+
+# Run a particular test.
+test='com.amazonaws.mobileconnectors.cognitoidentityprovider.SignInPersistenceIntegrationTest#testSignIn'
+./gradlew cAT -Pandroid.testInstrumentationRunnerArguments.class="$test"
+
+# Dump the device logs to your debugging file.
+adb logcat -d > device-logs-during-test.log
+```
+
+Now, you can inspect both the test execution results, as well as what was
+happening on the device, in `device-logs-during-test.log`.
+
+## Related Repositories
+
+This project is part of the Amplify Framework, which runs on Android,
+iOS, and numerous JavaScript-based web platforms.
+
+1. [AWS Amplify for iOS](https://github.com/aws-amplify/amplify-ios)
+2. [AWS Amplify for JavaScript](https://github.com/aws-amplify/amplify-js)
+
+AWS Amplify plugins are built on top of the AWS SDKs. AWS SDKs are a
+toolkit for interacting with AWS backend resources.
+
+1. [AWS SDK for Android](https://github.com/aws-amplify/aws-sdk-android)
+2. [AWS SDK for iOS](https://github.com/aws-amplify/aws-sdk-ios)
+3. [AWS SDK for JavaScript](https://github.com/aws/aws-sdk-js)
+
+Not officially part of the AWS SDKs, [AppSync](https://aws.amazon.com/appsync/) is an opinionated,
+mobile-oriented GraphQL management service. It is used by Amplify's
+DataStore and API plugins.
+
+1. [Android AppSync Client](https://github.com/awslabs/aws-mobile-appsync-sdk-android)
+2. [iOS AppSync Client](https://github.com/awslabs/aws-mobile-appsync-sdk-ios)
+3. [JavaScript AppSync Client](https://github.com/awslabs/aws-mobile-appsync-sdk-js)
```

Comment:
@palpatim (I think) suggested in the iOS one to just kill the AppSync links, worrying that people would get confused. I think that's right. We spent a lot of time on these SDKs internally since we are defacto the AppSync client team. But, these are largely irrelevant to the average AWS user.

Reply:
Agreed. I was confused by it. Removed.